src/1-foundations/1-type-theory/exercises/1.2-solution.rzk.md

```diff
@@ -6,6 +6,173 @@ This is a literate Rzk file:
 #lang rzk-1
 ```
 
-!!! warning
+Derive the recursion principle for products
+$rec_{A \times B}$ using only the projections,
+and verify that the definitional equalities are valid.
+Do the same for $\Sigma$-types.
 
-    To be done.
+## Solution
+
+### Product $A \times B$.
+
+Introducing definitions
+```rzk
+-- defined before
+
+-- #def id
+--   (A : U)
+--   : A -> A
+--   := \ a -> a
+
+-- #def prod
+--   (A B : U)
+--   : U
+--   := Sigma (_ : A), B
```

Comment:
Remove commented out code. You can refer to these definitions from text, e.g. something like this (I have not checked the validity of links):

```
In this solution, we use [`id`](../04-dependent-function-types.rzk/#define:id) and [`prod`](../05-product-types.rzk.md/#define:prod), defined earlier.
```